### PR TITLE
[14.0] [FIX] account_invoice_triple_discount: for in invoice_line_ids if exists, else on line_ids

### PR DESCRIPTION
**Pull request by odooNextev**
…ts, else on line_ids
fix https://github.com/OCA/account-invoicing/pull/1592#issuecomment-1807762328

### CODE REVIEW

**Comment by francesco-ooops** [comment]
cc @eLBati @sergiocorato

**Comment by francesco-ooops** [comment]
@OCA/accounting-maintainers this looks urgent to me

**Comment by francesco-ooops** [comment]
@SirAionTech what do you think of this one?

**Review by HekkiMelody — APPROVED** [review]
Code review LGTM

**chore**: please adjust the commit message following the guidelines: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#commit-message 

**suggestion (non-blocking)**: since this is a rather intricate and delicate matter (had to read multiple issues in multiple languages to understand the context), I think adding a test would be helpful in avoiding regressions and it would implicitly also explain the reason behind the change. If not a test, at least a comment.

**chore**: once approved, please FW to v15 and 16!

**Comment by francesco-ooops** [comment]
@ferran-S73 fyi

**Review by ferran-S73 — APPROVED** [review]

File: account_invoice_triple_discount/models/account_move.py
Comment (sergiocorato) [inline account_invoice_triple_discount/models/account_move.py:19]: I think it would be better to check if it `is_invoice` instead.

Comment (odooNextev) [reply account_invoice_triple_discount/models/account_move.py:19]: Working on the broken test, is_invoice returns True but without invoice_line_ids. So in this case it will not fix the test

**Comment by SirAionTech** [comment]
> @SirAionTech what do you think of this one?

I think there should be a test in this module that explains what this change is fixing and ensures that there are no regressions in the future.
Basically I agree with https://github.com/OCA/account-invoicing/pull/1605#pullrequestreview-1733844809, but I'm more strict :angel:.
Changing this module to fix tests of another module, _in another repo_ too, makes little sense in my opinion.

If this module ain't broken, don't fix it.

**Review by almumu — APPROVED** [review]

**Review by anmarmo1 — APPROVED** [review]
Functional review is ok.

**Review by eLBati — APPROVED** [review]
Thanks
(rif https://github.com/OCA/l10n-italy/pull/3681#discussion_r1401858443)

**Comment by eLBati** [comment]
> Changing this module to fix tests of another module, _in another repo_ too, makes little sense in my opinion

Why?

**Comment by SirAionTech** [comment]
> > Changing this module to fix tests of another module, _in another repo_ too, makes little sense in my opinion
> 
> Why?

Because in my opinion, a module should test its own behavior.
This avoids regressions: otherwise nothing would stop someone else from changing back this code: CI stays :green_circle:, so why not?

If triple discount works good for invoices, why is it breaking sales? I see a few options:
- Is **account_invoice_triple_discount** not working good? It should be reproducible with this module alone, so there should be a test in here.
- Is **sale_triple_discount** expecting something different from this module? Why? Maybe because it was relying on some wrong behavior of **account_invoice_triple_discount**? There might be something to fix in there.

**Comment by francesco-ooops** [comment]
@pedrobaeza can I ask your input here? 

the urgency is due to sale-workflow 14 tests currently failing

**Comment by eLBati** [comment]
> This avoids regressions: otherwise nothing would stop someone else from changing back this code: CI stays 🟢, so why not?

This makes sense.
@odooNextev did you already try to add a test that makes `account_invoice_triple_discount` fail without the changes of this PR?
Otherwise I can work on this next week.

**Comment by odooNextev** [comment]
> This makes sense. @odooNextev did you already try to add a test that makes `account_invoice_triple_discount` fail without the changes of this PR? Otherwise I can work on this next week.

We tried adding a test from `l10n_it_fatturapa_in` like this: https://github.com/OCA/l10n-italy/blob/14.0/l10n_it_fatturapa_in/tests/test_import_fatturapa_xml.py#L658

You can found it in the old PR: https://github.com/OCA/account-invoicing/pull/1442/files#diff-3c2a0c335b3658f70e6cc646d2f82097e866dbcd7267a0b3497df142d9cd2392R184

**Comment by pedrobaeza** [comment]
This patch sounds weird, but let's continue for unblocking the CI:

/ocabot merge patch

**Pull request merged**
